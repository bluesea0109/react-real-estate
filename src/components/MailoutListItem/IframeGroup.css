#cust-address-small {
  position: absolute;
  top: 59.45px;
  left: 26.5px;
  width: 75.67px;
  height: 37.83px;
  font-size: 7.57px;
  text-transform: uppercase;
  text-align: left;
}

#ink-free-area-small {
  border: 1px solid red;
  background-color: white;
  position: absolute;
  width: 1.7835in;
  height: 1.3035in;
  right: 0.05in;
  bottom: 0.05in;
}
#ink-free-text-small {
  position: absolute;
  bottom: 2.16px;
  right: 0.4px;
  font-size: 5.4px;
  text-transform: uppercase;
  width: 100%;
  text-align: center;
  color: darkred;
}
#postage-small {
  border: 1px solid #262626;
  width: 40.54px;
  height: 33.5px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  font-size: 6.48px;
  position: absolute;
  top: 0.125in;
  right: 0.2in;
}
