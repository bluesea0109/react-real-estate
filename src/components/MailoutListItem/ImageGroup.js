import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { Image } from '../Base';

const ImageGroup = ({ img1src, img2src, linkTo }) => {
  if (!img1src || !img2src) return;

  if (linkTo) {
    return (
      <Image.Group style={{ minWidth: '200px' }}>
        <Link to={linkTo}>
          <Image src={img1src} />
        </Link>
        <Link to={linkTo}>
          <Image src={img2src} />
        </Link>
      </Image.Group>
    );
  }

  return (
    <Image.Group size="big">
      <Image src={img1src} />
      <Image src={img2src} />
    </Image.Group>
  );
};

ImageGroup.propTypes = {
  img1src: PropTypes.string.isRequired,
  img2src: PropTypes.string.isRequired,
  size: PropTypes.string,
  linkTo: PropTypes.string,
};

export default ImageGroup;
