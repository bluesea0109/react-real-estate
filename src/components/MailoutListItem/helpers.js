import Dinero from 'dinero.js/build/esm/dinero.js';
import { format } from 'date-fns';

export const calculateCost = recipientCount => {
  if (!recipientCount || typeof recipientCount !== 'number') return '-';
  // The amount is represented in cents
  return Dinero({ amount: 59, currency: 'USD' })
    .multiply(recipientCount)
    .toFormat('$0,0.00');
};

export const canSend = mailoutStatus => {
  return mailoutStatus === 'calculated';
};

export const canPickDestinations = mailoutStatus => {
  return mailoutStatus === 'created';
}

export const formatDate = created => {
  if (!created || typeof created !== 'number') return '-';
  return format(new Date(created), 'MM/dd/yyyy');
};

export const resolveMailoutStatusIcon = mailoutStatus => {
  if (mailoutStatus === 'created') return 'edit';
  if (mailoutStatus === 'calculated') return 'eye';
  if (mailoutStatus === 'submitted') return 'sync-alt';
  // if (mailoutStatus === 'scheduled') return 'eye' // this has been sent to lob
  if (mailoutStatus === 'hide') return 'ban';
  if (mailoutStatus === 'archived') return 'archive';
  if (mailoutStatus === 'errored') return 'exclamation-triangle';
  if (mailoutStatus === 'cancelled') return 'times';

  return 'check';
};

export const resolveMailoutStatus = mailoutStatus => {
  if (mailoutStatus === 'created') return 'Created';
  if (mailoutStatus === 'calculated') return 'Awaiting Approval';
  if (mailoutStatus === 'submitted') return 'Processing';
  // if (mailoutStatus === 'scheduled') return 'Awaiting Approval' // this has been sent to lob
  if (mailoutStatus === 'hide') return 'Excluded';
  if (mailoutStatus === 'archived') return 'Archived';
  if (mailoutStatus === 'errored') return 'Errored';
  if (mailoutStatus === 'cancelled') return 'Canceled';

  return 'Sent';
};

export const resolveMailoutStatusColor = mailoutStatus => {
  if (mailoutStatus === 'created') return '#666666';
  if (mailoutStatus === 'calculated') return '#000000';
  if (mailoutStatus === 'submitted') return '#666666';
  // if (mailoutStatus === 'scheduled') return '#000000' // this has been sent to lob
  if (mailoutStatus === 'excluded') return '#555555';
  if (mailoutStatus === 'errored') return '#f2714d';
  if (mailoutStatus === 'cancelled') return '#f2714d';
  if (mailoutStatus === 'hide') return '#555555';
  if (mailoutStatus === 'archived') return 'grey';

  return '#59c4c4';
};

export const resolveLabelStatus = (listingStatus, mailoutStatus) => {
  if (mailoutStatus === 'hide') return 'grey';
  if (mailoutStatus === 'archived') return 'grey';
  if (listingStatus === 'sold') return 'orange';
  if (listingStatus === 'listed') return 'teal';

  return 'red';
};
