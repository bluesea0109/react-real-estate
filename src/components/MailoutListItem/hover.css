.bm-effect {
  transform: scale(1, 1);
  transition: 0.25s;
}

.bm-effect:hover {
  transform: scale(1.1, 1.1);
}
