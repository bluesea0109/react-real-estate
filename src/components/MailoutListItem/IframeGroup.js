import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { Segment } from 'semantic-ui-react';
import React, { useCallback, useState } from 'react';

import { iframeTransformMobile, iframeLinkStyle } from '../helpers';
import { ItemBodyIframeLayout } from '../../layouts';
import ApiService from '../../services/api/index';
import { Image } from '../Base';
import { useIsMobile } from '../Hooks/useIsMobile';

const IframeGroup = ({ index, item, linkTo = null }) => {

  const isMobile = useIsMobile();

  const peerId = useSelector(store => store.peer.peerId);

  const [frontLoaded, setFrontLoaded] = useState(false);
  const [backLoaded, setBackLoaded] = useState(false);

  const frontURL = peerId
    ? ApiService.directory.peer.mailout.render.front({ userId: item?.userId, peerId, mailoutId: item?._id }).path
    : ApiService.directory.user.mailout.render.front({ userId: item?.userId, mailoutId: item?._id }).path;

  const backURL = peerId
    ? ApiService.directory.peer.mailout.render.back({ userId: item?.userId, peerId, mailoutId: item?._id }).path
    : ApiService.directory.user.mailout.render.back({ userId: item?.userId, mailoutId: item?._id }).path;

  const handleOnload = useCallback(
    event => {
      const {
        name,
        document: { body },
      } = event.target.contentWindow;

      body.style.overflow = 'hidden';
      body.style['pointer-events'] = 'none';
      body.style.transform = iframeTransformMobile;

      if (name === 'front') {
        setFrontLoaded(true);
      }

      if (name === 'back') {
        setBackLoaded(true);
      }
    },
    [setFrontLoaded, setBackLoaded]
  );

  const imgStyles = {
    maxWidth: '300px',
    minWidth: '290px',
    height: '204px',
    overflow: 'hidden',
    marginLeft: 0,
  }

  if (linkTo) {
    return (
      <ItemBodyIframeLayout horizontal={!isMobile} style={{ border: 'none', boxShadow: 'none' }} id={`mailout-iframe-set-${index}`}>
      {item.frontResourceUrl && (
        <Segment textAlign="center" style={{ border: 'none' }}>

          <Link to={linkTo}>
            <Image src={item.frontResourceUrl} style={imgStyles} className="bm-transform-effect image-frame-border" />
          </Link>
        </Segment>
      )}
      {!item.frontResourceUrl && (
        <Segment textAlign="center" loading={!item?._id || !frontLoaded} style={{ border: 'none' }}>
          <div style={{ width: '300px', height: '204px', overflow: 'hidden' }} className="bm-transform-effect image-frame-border">
            <iframe
              id="bm-iframe-front"
              title={frontURL + '?dashboard=true'}
              name="front"
              src={null}
              width="300"
              height="204"
              frameBorder="none"
              sandbox="allow-same-origin allow-scripts"
              onLoad={handleOnload}
              style={{ visibility: !item?._id || !frontLoaded ? 'hidden' : 'visible' }}
            />
            <Link to={linkTo} style={iframeLinkStyle} />
          </div>
        </Segment>
      )}


        <Segment textAlign="center" loading={!item?._id || !backLoaded} style={{ border: 'none' }}>
          <div style={{ width: '300px', height: '204px', overflow: 'hidden' }} className="bm-transform-effect image-frame-border">
            <iframe
              id="bm-iframe-back"
              title={backURL + '?dashboard=true'}
              name="back"
              src={null}
              width="300"
              height="204"
              frameBorder="none"
              sandbox="allow-same-origin allow-scripts"
              onLoad={handleOnload}
              style={{ visibility: !item?._id || !backLoaded ? 'hidden' : 'visible' }}
            />
            <Link to={linkTo} style={iframeLinkStyle} />
          </div>
        </Segment>
      </ItemBodyIframeLayout>
    );
  } else {
    return (
      <ItemBodyIframeLayout horizontal={!isMobile} style={{ border: 'none', boxShadow: 'none' }} id={`mailout-iframe-set-${index}`}>
        <Segment textAlign="center" loading={!item?._id || !frontLoaded} style={{ border: 'none' }}>
          <div style={{ width: '300px', height: '204px', overflow: 'hidden' }}>
            <iframe
              id="bm-iframe-front"
              title={frontURL}
              name="front"
              src={null}
              width="300"
              height="204"
              frameBorder="none"
              sandbox="allow-same-origin allow-scripts"
              onLoad={handleOnload}
              style={{ visibility: !item?._id || !frontLoaded ? 'hidden' : 'visible' }}
            />
          </div>
        </Segment>
        <Segment textAlign="center" loading={!item?._id || !backLoaded} style={{ border: 'none' }}>
          <div style={{ width: '300px', height: '204px', overflow: 'hidden' }}>
            <iframe
              id="bm-iframe-back"
              title={backURL}
              name="back"
              src={null}
              width="300"
              height="204"
              frameBorder="none"
              sandbox="allow-same-origin allow-scripts"
              onLoad={handleOnload}
              style={{ visibility: !item?._id || !backLoaded ? 'hidden' : 'visible' }}
            />
          </div>
        </Segment>
      </ItemBodyIframeLayout>
    );
  }
};

IframeGroup.propTypes = {
  item: PropTypes.object.isRequired,
  linkTo: PropTypes.string,
};

export default IframeGroup;
