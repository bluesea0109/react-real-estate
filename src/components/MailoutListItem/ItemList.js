import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { calculateCost, formatDate, resolveMailoutStatus, resolveMailoutStatusColor, resolveMailoutStatusIcon } from './helpers';
import { ItemBodyDataLayout } from '../../layouts';
import PopupContent from './PopupContent';
import { List, Popup } from '../Base';
import { postcardDimensionsDisplayed } from '../utils';
import './styles/itemList.scss'

const ItemList = ({ data }) => {
  if (!data) return;

  return (
    <div id="listingsItems" style={{padding: '18px 30px 0px 18px'}}>
      <ItemBodyDataLayout relaxed>
        <List.Item>
          <List.Content>
            <List.Header>Recipients</List.Header>
            <List.Description>{data.recipientCount}</List.Description>
          </List.Content>
        </List.Item>
        <List.Item>
          <List.Content>
            <List.Header>Size</List.Header>
            <List.Description>{data.postcardSize ? `${postcardDimensionsDisplayed(data.postcardSize)}"` : '4x6"'}</List.Description>
          </List.Content>
        </List.Item>
        <List.Item>
          <List.Content>
            <List.Header>Cost</List.Header>
            <List.Description>{calculateCost(data.recipientCount, data.postcardSize ? data.postcardSize : '4x6')}</List.Description>
          </List.Content>
        </List.Item>
        <List.Item>
          <List.Content>
            <List.Header>
              Status
              <Popup
                flowing
                trigger={<FontAwesomeIcon icon="info-circle" style={{ marginLeft: '.5em', color: '#2DB5AD' }} />}
                content={PopupContent()}
                position="top right"
              />
            </List.Header>
            <List.Description style={{ padding: '1em 0em 1em 0em', color: resolveMailoutStatusColor(data.mailoutStatus) }}>
              <FontAwesomeIcon icon={resolveMailoutStatusIcon(data.mailoutStatus)} style={{ marginRight: '.5em' }} />
              {resolveMailoutStatus(data.mailoutStatus)}
            </List.Description>
          </List.Content>
        </List.Item>
        <List.Item>
          <List.Content>
            <List.Header>Created</List.Header>
            <List.Description>{formatDate(data.created)}</List.Description>
          </List.Content>
        </List.Item>
      </ItemBodyDataLayout>
    </div>
  );
};

ItemList.propTypes = {
  data: PropTypes.object.isRequired,
};

export default ItemList;
