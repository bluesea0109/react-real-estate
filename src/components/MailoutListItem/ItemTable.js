import React from 'react';
import PropTypes from 'prop-types';
import { Popup, Table } from 'semantic-ui-react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { calculateCost, formatDate, resolveMailoutStatus, resolveMailoutStatusColor, resolveMailoutStatusIcon } from './helpers';
import PopupContent from './PopupContent';

const ItemTable = ({ data }) => {
  if (!data) return;

  return (
    <Table basic="very">
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>Recipients</Table.HeaderCell>
          <Table.HeaderCell>Cost</Table.HeaderCell>
          <Table.HeaderCell>
            Status
            <Popup
              flowing
              trigger={<FontAwesomeIcon icon="info-circle" style={{ marginLeft: '.5em', color: '#2DB5AD' }} />}
              content={PopupContent()}
              position="top right"
            />
          </Table.HeaderCell>
          <Table.HeaderCell>Created</Table.HeaderCell>
        </Table.Row>
      </Table.Header>

      <Table.Body>
        <Table.Row>
          <Table.Cell>{data.recipientCount}</Table.Cell>
          <Table.Cell>{calculateCost(data.recipientCount)}</Table.Cell>
          <Table.Cell style={{ color: resolveMailoutStatusColor(data.mailoutStatus) }}>
            <FontAwesomeIcon icon={resolveMailoutStatusIcon(data.mailoutStatus)} style={{ marginRight: '.5em' }} />
            {resolveMailoutStatus(data.mailoutStatus)}
          </Table.Cell>
          <Table.Cell>{formatDate(data.created)}</Table.Cell>
        </Table.Row>
      </Table.Body>
    </Table>
  );
};

ItemTable.propTypes = {
  data: PropTypes.object.isRequired,
};

export default ItemTable;
