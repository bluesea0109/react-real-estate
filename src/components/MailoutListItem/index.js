import React from 'react';
import PropTypes from 'prop-types';

import { ItemLayout, ItemBodyLayout } from '../../layouts';
import ImageGroup from './ImageGroup';
import ListHeader from './ListHeader';
import ItemList from './ItemList';

const MailoutListItem = data => {
  if (!data) return;

  return (
    <ItemLayout fluid key={`${data.userId}-${data._id}-${data.mlsNum}`}>
      {ListHeader({ data: data })}
      <ItemBodyLayout attached style={{ padding: 10 }}>
        {ImageGroup({
          img1src: data.sampleBackLargeUrl,
          img2src: data.sampleFrontLargeUrl,
          linkTo: `dashboard/${data._id}`,
        })}

        {ItemList({ data: data })}
      </ItemBodyLayout>
    </ItemLayout>
  );
};

MailoutListItem.propTypes = {
  data: PropTypes.object.isRequired,
};

export default MailoutListItem;
