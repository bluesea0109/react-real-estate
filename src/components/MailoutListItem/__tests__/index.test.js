'use strict';

import matchMedia from '../../../__mocks__/matchMedia';
import mailoutsData from '../../../__mocks__/mailouts';

import React from 'react';
import { MemoryRouter } from 'react-router';
import MailoutListItem from '../';
import renderer from 'react-test-renderer';

const data = mailoutsData[0];

it('renders correctly', () => {
  const tree = renderer.create(<MemoryRouter>{MailoutListItem(data)}</MemoryRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
