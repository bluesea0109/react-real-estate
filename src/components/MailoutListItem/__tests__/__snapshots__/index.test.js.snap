// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui fluid container sc-dnqmqq hrZLex"
>
  <div
    className="ui block top attached header sc-gzVnrw ecvOFU"
  >
    <span
      style={
        Object {
          "gridArea": "label",
        }
      }
    >
      <div
        className="ui teal large ribbon label"
        onClick={[Function]}
        style={
          Object {
            "left": "-2.7em",
            "textTransform": "capitalize",
            "top": "-0.9em",
          }
        }
      >
        listed
      </div>
    </span>
    <span
      style={
        Object {
          "alignSelf": "center",
          "gridArea": "address",
        }
      }
    >
      <a
        className="ui header"
        href="/dashboard/ck34pkmza003t01qwnqwnyg4d"
        onClick={[Function]}
      >
        <h3
          className="sc-kgoBCf kVSTZj bm-color-effect"
        >
          1305 Thibodeau Ln
        </h3>
      </a>
    </span>
    <span
      className="sc-htoDjs cnqDiZ"
    >
      <span />
      <span />
      <span />
    </span>
  </div>
  <div
    className="ui attached segment sc-iwsKbI gVIoui"
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <div
      className="ui sc-gqjmRU hALeKB images"
    >
      <a
        href="/dashboard/ck34pkmza003t01qwnqwnyg4d"
        onClick={[Function]}
      >
        <img
          className="ui image sc-kpOJdX kAzjOl bm-transform-effect"
          src="https://lob-assets.com/postcards/psc_c7c818dae0be2c24_thumb_large_2.png?version=v1&expires=1576690772&signature=Slb3d3xMzrMAiJVNRW4yHEhDIZbgK33oXqWwn-3ytIaVk8gXsWEYtBE_otaiFFoc1H7h04REiFpZfRoKlEYGDg"
          style={
            Object {
              "width": "48%",
            }
          }
        />
      </a>
      <a
        href="/dashboard/ck34pkmza003t01qwnqwnyg4d"
        onClick={[Function]}
      >
        <img
          className="ui image sc-kpOJdX kAzjOl bm-transform-effect"
          src="https://lob-assets.com/postcards/psc_c7c818dae0be2c24_thumb_large_1.png?version=v1&expires=1576690772&signature=zoqOEZ5XXSL0MPWivtWrmri0B_JEsvKOh2tcXPukiKG3U_tRwibbz9wmTzZOL17rr-4RRWOJMB3rg06hEfOADw"
          style={
            Object {
              "width": "48%",
            }
          }
        />
      </a>
    </div>
    <div
      className="ui relaxed list sc-VigVT jDcUJV"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            Recipients
          </div>
          <div
            className="description"
          >
            100
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            Cost
          </div>
          <div
            className="description"
          >
            $59.00
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            Status
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-info-circle fa-w-16 "
              data-icon="info-circle"
              data-prefix="fas"
              focusable="false"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="img"
              style={
                Object {
                  "color": "#2DB5AD",
                  "marginLeft": ".5em",
                }
              }
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <div
            className="description"
            style={
              Object {
                "color": "#59c4c4",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-check fa-w-16 "
              data-icon="check"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "marginRight": ".5em",
                }
              }
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Sent
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            Created
          </div>
          <div
            className="description"
          >
            11/18/2019
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
